src/utils/axiosRequest.ts

```diff
@@ -0,0 +1,31 @@
+import axios, { AxiosRequestConfig, Method } from "axios";
+
+// create an axios instance
+const service = axios.create({
+  baseURL: "api",
+  timeout: process.env.REQUEST_TIMEOUT as number | undefined,
+  withCredentials: true,
+});
+
+// resquest interceptor - before request is sent
+service.interceptors.request.use((config) => {
+  const accessToken = localStorage.getItem("accessToken");
+  accessToken && (config.headers["Authorization"] = `Bearer ${accessToken}`);
+  return config;
+});
+
+const axiosRequest = async(
+  url: string,
+  method: Method,
+  submitData: object,
+  config?: AxiosRequestConfig
+) => {
+  return await service.request({
+    url,
+    method,
+    [method.toLocaleLowerCase() === "get" ? "params" : "data"]: submitData,
```

Comment:
You should restructure the code of axiosRequest.
Please refer this article.
https://juejin.cn/post/7090889657721815076